src/data/fintype/basic.lean

```diff
@@ -1591,6 +1591,15 @@ not_nonempty_iff.trans is_empty_fintype
 noncomputable def fintype_of_not_infinite {α : Type*} (h : ¬ infinite α) : fintype α :=
 ((not_iff_comm.mp not_nonempty_fintype).mp h).some
 
+section
+open_locale classical
+
+/-- Any type is (classically) either a `fintype`, or `infinite`. -/
```

Comment:
It might be nice to mention the example use of `cases fintype_or_infinite α; resetI` (it wasn't immediately clear to me why it was useful.)

Reply:
Done.